projects/dg-docassembly-webcomponent/src/lib/assembly/service/assembly.service.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { catchError, map } from 'rxjs/operators';
+
+const uiDefinitionEndpoint = `api/form-definitions`;
+const generateDocumentEndpoint = `api/template-renditions`;
+
+export const  ERROR = '--error--';
+
+@Injectable()
+export class AssemblyService {
+
+  constructor(private http: HttpClient) {}
+
+  getUIDefinition(templateName: string): Observable<FormlyFieldConfig[]> {
+    const encTemplateId = btoa(templateName);
+    return this.http.get<any>(`${uiDefinitionEndpoint}/${encTemplateId}`)
```

Comment:
one day I would like to stop doing <any> for the server responses, it shouldn't be that hard to create an interface for them. I understand that this just how it's done everywhere though